File: 219-ContainsDuplicateIi/219-ContainsDuplicateIi.java
// Last updated: 8/1/2025, 7:09:58 AM
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        for (int i=0;i<nums.length;i++){
            for (int j=i+1;j<nums.length && j<=i+k;j++){
                if (nums[i]==nums[j]){
                    return true;
                }
            }
        }
        return false;
    }
}

File: 219-ContainsDuplicateII/219-ContainsDuplicateII.cpp
// Last updated: 2/26/2026, 11:13:00 AM
1class Solution {
2public:
3    bool containsNearbyDuplicate(vector<int>& nums, int k) {
4        unordered_map<int,int> s;
5        int n = nums.size();
6        for (int i=0 ; i<n ; i++){
7            int num = nums[i];
8            if (s.find(num)!=s.end()){
9                int prevIdx = s[num];
10                if (i-prevIdx<=k){
11                    return true;
12                }
13            }
14            s[num] = i;
15        }
16        return false;
17    }
18};